File: javascript/hamming/hamming.js
`use strict`
var Hamming = function () {}

Hamming.prototype.compute = function (string1, string2) {
  count = 0
  if (string1.length !== string2.length) {
    throw Error('DNA strands must be of equal length.')
  }
  for (var i = 0; i < string1.length; i++) {
    if (string1.charAt(i) !== string2.charAt(i)) {
      count += 1
    }
  }
  return count
}
module.exports = Hamming

File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(string1, string2)
    if string1.length != string2.length
      raise ArgumentError
    end
    i = 0
    count = 0
    while i < string1.length
      if string1[i] != string2[i]
        count += 1
      end
      i += 1
    end
    count
  end
end

module BookKeeping
  VERSION = 3
end
